src/Renci.SshNet/Common/AsyncResult`1.cs

```diff
@@ -6,7 +6,9 @@ namespace Renci.SshNet.Common
     /// Base class to encapsulates the results of an asynchronous operation that returns result.
     /// </summary>
     /// <typeparam name="TResult">The type of the result.</typeparam>
+#pragma warning disable SA1649 // Normal convention is to name generic files `1, naming this AsyncResult{TResult}.cs breaks docfx.
     public abstract class AsyncResult<TResult> : AsyncResult
+#pragma warning restore SA1649
```

Comment:
It looks like you renamed the file, so presumably the suppression is not needed now

```suggestion
    public abstract class AsyncResult<TResult> : AsyncResult
```

Reply:
If you try it, you'll notice that as soon as you remove the warning, it'll tell you to rename the file back to `AsyncResult{TResult}.cs` (which I've never seen to be honest), and it'll break docfx.

Reply:
Right. I assumed it had been telling you to rename to `1. This fixes it:

```diff
diff --git a/stylecop.json b/stylecop.json
index ff7c9dbf..ff643f45 100644
--- a/stylecop.json
+++ b/stylecop.json
@@ -3,7 +3,9 @@
     "settings": {
         "documentationRules": {
             "xmlHeader": false,
-            "documentInternalElements": false
+            "documentInternalElements": false,
+            "fileNamingConvention": "metadata"
         },
         "layoutRules": {
             "newlineAtEndOfFile": "require"
```

Reply:
Indeed it did, thanks.

Reply:
I removed the warning as well.